src/sagemaker/workflow/fail_step.py

```diff
@@ -36,14 +36,15 @@ def __init__(
         """Constructs a FailStep.
 
         Args:
-            name (str): The name of the Fail step.
-            error_message (str or PipelineNonPrimitiveInputTypes): Error message defined by user.
-                Once the Fail step is reached the execution will fail and the
+            name (str): The name of the fail step. A name is required and must be unique within an account.
+            error_message (str or PipelineNonPrimitiveInputTypes): An error message defined by the user.
+                Once the fail step is reached, the execution will fail and the
                 error message will be set as the failure reason (default: None).
-            display_name (str): The display name of the Fail step (default: None).
-            description (str): The description of the Fail step (default: None).
+            display_name (str): The display name of the fail step. The display name provides better readability 
+                for fail step naming conventions that are auto-generated with random numbers or timestamps. (default: None).
```

Comment:
`display_name (str): The display name of the fail step. The display name can provide better readability in UI. (default: None)` is good enough. 

The step name is not necessary to be always auto-generated with random numbers/ timestamps. In a small pipeline with only limited steps, users usually manually define the step name.

Reply:
- This parameter is inherited from FailStep's parent class [Step](https://github.com/aws/sagemaker-python-sdk/blob/dev/src/sagemaker/workflow/steps.py#L67). `display_name` is not required but it can help in some special cases like if the step name it self is not readable
- It ban be assigned directly from the constructor